File: liband/src/main/java/com/liband/utils/SnackbarUtils.java
package com.liband.utils;

import android.graphics.Color;
import android.view.View;
import android.widget.TextView;

import com.google.android.material.snackbar.Snackbar;
import com.liband.R;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;


public class SnackbarUtils {

    private static Snackbar snackbar;

    public static void loadSnackBar(String message, AppCompatActivity activity, int color) {
        if (activity != null) {
            snackbar = Snackbar.make(
                    activity.findViewById(android.R.id.content), message,
                    Snackbar.LENGTH_LONG);

//              .setAction(
//                    activity.getResources().getString(R.string.ok),
//                    new OnClickListener() {
//                        @Override
//                        public void onClick(View v) {
//                        }
//                    })

//            snackbar.setActionTextColor(activity.getResources().getColor(
//                    R.color.white));
            View snackbarView = snackbar.getView();
            final TextView tv = snackbarView.findViewById(R.id.snackbar_text);
            tv.setTextColor(Color.WHITE);
            final int textSize = (int) activity.getResources().getDimension(R.dimen._5ssp);
            tv.setTextSize(textSize);
            snackbarView.setBackgroundColor(ContextCompat.getColor(activity, color));
            snackbar.show();
        }
    }


    public static void loadSnackBar(final View view, String message, AppCompatActivity activity, int color) {
        if (activity != null) {
            snackbar = Snackbar.make(
                    view, message,
                    Snackbar.LENGTH_LONG);

//              .setAction(
//                    activity.getResources().getString(R.string.ok),
//                    new OnClickListener() {
//                        @Override
//                        public void onClick(View v) {
//                        }
//                    })

//            snackbar.setActionTextColor(activity.getResources().getColor(
//                    R.color.white));
            View snackbarView = snackbar.getView();
            final TextView tv = snackbarView.findViewById(R.id.snackbar_text);
            tv.setTextColor(Color.WHITE);
            final int textSize = (int) activity.getResources().getDimension(R.dimen._5ssp);
            tv.setTextSize(textSize);
            snackbarView.setBackgroundColor(ContextCompat.getColor(activity, color));
            snackbar.show();
        }
    }


    public static void cancelSnackBar() {
        if (snackbar != null)
            snackbar.dismiss();
    }


}

File: liband/src/main/java/com/liband/utils/SnackbarUtils.kt
package com.liband.utils

import android.app.Activity
import android.graphics.Color
import android.view.View
import android.widget.TextView
import androidx.core.content.ContextCompat
import com.google.android.material.snackbar.Snackbar
import com.liband.R


object SnackbarUtils {

    private var snackbar: Snackbar? = null

    fun loadSnackBar(message: String, activity: Activity?, color: Int) {
        if (activity != null) {
            snackbar = Snackbar.make(
                    activity.findViewById(android.R.id.content), message,
                    Snackbar.LENGTH_LONG)

            //              .setAction(
            //                    activity.getResources().getString(R.string.ok),
            //                    new OnClickListener() {
            //                        @Override
            //                        public void onClick(View v) {
            //                        }
            //                    })

            //            snackbar.setActionTextColor(activity.getResources().getColor(
            //                    R.color.white));
            val snackbarView = snackbar!!.view
            val tv = snackbarView.findViewById<TextView>(R.id.snackbar_text)
            tv.setTextColor(Color.WHITE)
            val textSize = activity.resources.getDimension(R.dimen._5ssp).toInt()
            tv.textSize = textSize.toFloat()
            snackbarView.setBackgroundColor(ContextCompat.getColor(activity, color))
            snackbar!!.show()
        }
    }


    fun loadSnackBar(view: View, message: String, activity: Activity?, color: Int) {
        if (activity != null) {
            snackbar = Snackbar.make(
                    view, message,
                    Snackbar.LENGTH_LONG)

            //              .setAction(
            //                    activity.getResources().getString(R.string.ok),
            //                    new OnClickListener() {
            //                        @Override
            //                        public void onClick(View v) {
            //                        }
            //                    })

            //            snackbar.setActionTextColor(activity.getResources().getColor(
            //                    R.color.white));
            val snackbarView = snackbar!!.view
            val tv = snackbarView.findViewById<TextView>(R.id.snackbar_text)
            tv.setTextColor(Color.WHITE)
            val textSize = activity.resources.getDimension(R.dimen._5ssp).toInt()
            tv.textSize = textSize.toFloat()
            snackbarView.setBackgroundColor(ContextCompat.getColor(activity, color))
            snackbar!!.show()
        }
    }


    fun cancelSnackBar() {
        if (snackbar != null)
            snackbar!!.dismiss()
    }


}
